kousa/lib/kousa/user.ex

```diff
@@ -75,4 +75,52 @@ defmodule Kousa.User do
       _ -> {:error, "tried to ban #{user_id_to_ban} but that user didn't exist"}
     end
   end
+
+  def admin_update_with(username_to_change, staff, contributions, opts) do
+    authorized_github_id = Application.get_env(:kousa, :ben_github_id, "")
+
+    with %{githubId: ^authorized_github_id} <- Users.get_by_id(opts[:admin_id]) do
+      user_to_change = Users.get_by_username(username_to_change)
+
+      cond do
+        not is_nil(staff) and not is_nil(contributions) ->
+          case user_to_change
```

Comment:
admin_update_with should function like update_with where you just pass the changeset in, you don't need case statements

```
user_to_change = Users.get_by_username(username_to_change)
```
^that should go in initialize
```
  def initialize(state) do
    Users.get_by_username(username_to_change)
  end
  ```

This might be tricky for you to figure out, I'll convert it over if you get stuck